apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Trace/TraceDecisionPredicate.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright 2017 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+
+#if NET45
+using System.Web;
+#else
+using Microsoft.AspNetCore.Http;
+#endif
+
+namespace Google.Cloud.Diagnostics.Common
+{
+    /// <summary>
+    /// Used to determine if an <see cref="HttpRequest"/> should be traced. This will not override
+    /// a decision by the trace header. 
+    /// </summary>
+    public class TraceDecisionPredicate
+    {
+        private readonly Func<HttpRequest, bool?> _traceDecisionPredicate;
+        private readonly bool _ignoreHealthChecks;
+
+        private TraceDecisionPredicate(
+            Func<HttpRequest, bool?> traceDecisionPredicate, bool ignoreHealthChecks)
+        {
+            _traceDecisionPredicate = traceDecisionPredicate;
+            _ignoreHealthChecks = ignoreHealthChecks;
+        }
+
+        /// <summary>
+        /// A <see cref="TraceDecisionPredicate"/> that does not have any affect on
```

Comment:
Nit: affect => effect.
And maybe be specific that health-checks are *not* disabled by this?
And maybe make it a property?